sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuWindowExpression.scala

```diff
@@ -185,8 +185,12 @@ case class GpuWindowExpression(windowFunction: Expression, windowSpec: GpuWindow
     val totalExtraColumns = numGroupingColumns
 
     val aggColumn = withResource(GpuProjectExec.project(cb, boundRowProjectList)) { projected =>
-      withResource(GpuColumnVector.from(projected)) { table =>
-        val bases = GpuColumnVector.extractBases(projected).zipWithIndex
+
+      // in case boundRowProjectList is empty
+      val finalCb = if (boundRowProjectList.length > 0) projected else cb
```

Comment:
curious why this is a special case for `WindowExpression` as opposed to be the general behavior of `GpuProjectExec#project` in which case the logic should live there?

Reply:
Do we really want `GpuProject` to return the original batch if the project expression is degenerate?  I'm not convinced that's true across the board.

Reply:
We may check the behavior of Spark project.

Reply:
I just tested below code

``` scala
    val unsafeProj = UnsafeProjection.create(Seq())
    val x = new GenericInternalRow(3)
    x.update(0, 1)
    x.update(1, 10)
    x.update(2, 100)
    println(s"x = ${x}")
    val y  = unsafeProj.apply(x)
    println(s"y = ${y}")
```

the output 
``` console
x = [1,10,100]
y = []
```
Seems if the project list is empty, then the output row will be empty.